internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,142 @@
+package entitysources
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/operator-registry/alpha/property"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource map[deppy.Identifier]input.Entity
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) (catalogdEntitySource, error) {
+	entities := catalogdEntitySource{}
+	bundleMetadatas, packageMetdatas, err := onClusterMetadata(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+	for _, bundle := range bundleMetadatas.Items {
+		props := map[string]string{}
+
+		for _, prop := range bundle.Spec.Properties {
+			switch prop.Type {
+			case property.TypePackage:
+				type pkg struct {
+					Name    string `json:"packageName"`
+					Version string `json:"version"`
+				}
+				tmp := pkg{}
+				if err := json.Unmarshal(prop.Value, &tmp); err != nil {
+					return nil, err
+				}
+				pkgValue, err := jsonMarshal(tmp)
+				if err != nil {
+					return nil, err
+				}
+				props[property.TypePackage] = string(pkgValue)
+			}
+		}
+
+		imgValue, err := jsonMarshal(bundle.Spec.Image)
+		if err != nil {
+			return nil, err
+		}
+		props["olm.bundle.path"] = string(imgValue)
+		bundlePkg := packageMetdatas[bundle.Spec.Package]
+		channelName := ""
+		for _, ch := range bundlePkg.Spec.Channels {
+			for _, b := range ch.Entries {
+				if b.Name == bundle.Name {
+					channelName = ch.Name
+				}
+			}
+		}
+		channelValue, _ := jsonMarshal(property.Channel{ChannelName: channelName, Priority: 0})
+		props[property.TypeChannel] = string(channelValue)
+		entity := input.Entity{
+			ID:         deppy.IdentifierFromString(fmt.Sprintf("%s%s%s", bundle.Name, bundle.Spec.Package, channelName)),
+			Properties: props,
+		}
+		entities[entity.ID] = entity
```

Comment:
This isn't the greatest, but since we have work planned for in the catalogd side that'll change these APIs, I didn't optimize this at all, and only included the bare minimum that deppy reconciles with today (I.e did not include any properties other than package name, version, bundle image and channel it belongs to)